File: TypeScript/jest/src/africanParrot.ts
import {Parrot, ParrotTypes} from './parrot';

class AfricanParrot extends Parrot {
    constructor(numberOfCoconuts: number, voltage: number, isNailed: boolean) {
        super(ParrotTypes.AFRICAN, numberOfCoconuts, voltage, isNailed);
    }

    getSpeed(): number {
        return super.getSpeed();
    }
}

export default AfricanParrot;

File: Java-2/src/main/java/parrot/AfricanParrot.java
package parrot;

public class AfricanParrot extends Parrot {
    public AfricanParrot(ParrotTypeEnum _type, int numberOfCoconuts, double voltage, boolean isNailed) {
        super(_type, numberOfCoconuts, voltage, isNailed);
    }

    public double getSpeed() {
        return Math.max(0, getBaseSpeed() - getLoadFactor() * numberOfCoconuts);
    }

    private double getLoadFactor() {
        return 9.0;
    }

}
